parser/pageparser/pagelexer_intro.go

```diff
@@ -155,7 +155,13 @@ LOOP:
 		}
 	}
 
+	// The front matter that was just lexed is in a format native to Org mode - it is,
+	// contrary to other document and front matter formats, part of the main section.
+	// l.emit advances the offset (start) of the lexer (and thus excludes the front matter
+	// from the main section) so we have to reset it.
+	frontMatterStart := l.start
```

Comment:
This doesn't look right. I don't understand the comment, but even if I did: This will produce more content than it consumes, and I'm fairly confident that this can be solved without this workaround.

Reply:
I'll try to my goal, maybe you can point me into a direction:
I want hugo to send the complete file contents to go-org - not just what hugo calls the MainSection part after the front matter.

---- 
The reason for this is that the front matter possibly contains information that is important for the rendering as `#+FOO: bar` is the [settings format of the org-mode parser](https://orgmode.org/manual/In_002dbuffer-settings.html).

An example:
```
#+TITLE: Foo
#+TODO: TODO DONE CUSTOM

content [...]
```
 
Hugo currently only sends  
```
content [...]
```
to the renderer. However, the `#+TODO: TODO DONE CUSTOM` is actually configuration for the Org mode renderer - `go-org` would use it to update the list of keywords that are recognized as TODO keywords in headlines - see [the headline with CUSTOM todo status example](https://niklasfasching.github.io/go-org/#headlines.org).

Reply:
OK; I think I understand. What you get today is a processed version of the content (after front matter). If you also need the front matter, I suggest that you add that to the `RenderingContext`. You can easily get to that with the parse tree you have.

Reply:
You could store away the start/end positions for front matter here:

https://github.com/gohugoio/hugo/blob/master/hugolib/page_content.go#L92

We currently don't use it.

Reply:
>So I guess the problem this change is trying to solve is that the front matter of an org file actually serves two purposes: (1) configure hugo (2) configure the Org mode parser.

Why not send the unmarshaled frontmatter?

https://github.com/gohugoio/hugo/blob/master/hugolib/page_content.go#L94

Reply:
Thanks! this sounds like it will get more involved than a 2 line change. 

As the benefit of this is pretty small for now (there's not much to be configured in go-org for now) i suggest we split this into a separate PR and just focus on org mode support itself for now.

Reply:
>As the benefit of this is pretty small for now

Agree. And Hugo's API in this area is also rather clumsy. I have been planning on redoing that for a while ... Bu time.

Reply:
Also note that If you need some "configuration for the rendererd", we should add that as configuration in Hugo which you could receive in some constructor.

Reply:
> Also note that If you need some "configuration for the rendererd", we should add that as configuration in Hugo which you could receive in some constructor.

Maybe I shouldn't have called it configuration for the renderer. It's not configuration in the sense that it should be passed to the constructor. Rather, we currently remove part of the org document because we conflate hugo configuration (front matter) and org mode configuration (buffer settings) by using the same syntax (`#+KEY: VALUE`), location (top of the file) and namespace for both of them. So... hugo remove front matter from the document to be rendered because that's the right thing - the problem with org is that the front matter is not just front matter - it's also part of the actual document.

Coming back to the point before
> Why not send the unmarshaled frontmatter?
I like the idea because it sounds like a good way to provide go-org with the full document without polluting the hugo code base with org specific stuff - having the front matter available sounds like it could be useful for other things as well.

I hope that makes sense...